naptime/src/main/scala/org/coursera/naptime/Types.scala

```diff
@@ -91,34 +92,39 @@ object Types extends StrictLogging {
       case unknown: DataSchema =>
         throw new RuntimeException(s"Cannot compute asymmetric type for key type: $unknown")
     }
-    for ((name, graphQLRelation) <- fields.graphQLRelations) {
-      Option(mergedSchema.getField(name)) match {
-        case Some(field) =>
-          logger.warn(
-            s"Fields for resource $typeName tries to add graphql relation on field " +
-              s"'$name', but that field is already defined on the model")
-        case None =>
-          val errorMessageBuilder = new StringBuilder
-          val newField = graphQLRelation.toAnnotation.relationType match {
-            case FINDER | MULTI_GET =>
-              val newField = new RecordDataSchema.Field(new ArrayDataSchema(new StringDataSchema)) // TODO(bryan): fix type here
-              newField.setOptional(false)
-              newField
-            case SINGLE_ELEMENT_FINDER | GET =>
-              val newField = new RecordDataSchema.Field(new StringDataSchema) // TODO(bryan): fix type here
-              newField.setOptional(true)
-              newField
-            case _ =>
-              throw new RuntimeException(
-                "unknown relation type: " +
-                  graphQLRelation.toAnnotation.relationType.toString)
-          }
-          val relationMap = Map[String, AnyRef](
-            Relations.RELATION_PROPERTY_NAME -> graphQLRelation.toAnnotation.data())
-          newField.setProperties(relationMap.asJava)
-          newField.setDoc(graphQLRelation.description)
-          newField.setName(name.split("/").last, errorMessageBuilder)
-          insertFieldAtLocation(mergedSchema, name.split("/").dropRight(1).toList, newField)
+    val relationKeys = fields.relations.keys ++ fields.graphQLRelations.keys
```

Comment:
nit: should this be distincted? it could be that they are already both sets so this is not a problem but it's not clear just reading this code.

Reply:
Yes, I just noticed this as well -- it should be distincted.